agent/app/repo/app.go

```diff
@@ -76,6 +77,12 @@ func (a AppRepo) WithResource(resource string) DBOption {
 	}
 }
 
+func (a AppRepo) WithNotLocal() DBOption {
+	return func(g *gorm.DB) *gorm.DB {
+		return g.Where("resource != local")
+	}
+}
+
 func (a AppRepo) WithArch(arch string) DBOption {
 	return func(g *gorm.DB) *gorm.DB {
 		return g.Where("architectures like ?", fmt.Sprintf("%%%s%%", arch))
```

Comment:
There are no significant differences between the two versions of the code. The implementation remains consistent and does not require any changes to achieve the desired functionality.